lib/dal/src/prompt_override.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{DalContext, WsEvent, WsEventResult, WsPayload};
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use telemetry::prelude::*;
+use thiserror::Error;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum PromptOverrideError {
+    #[error("pg error: {0}")]
+    Pg(#[from] si_data_pg::PgError),
+    #[error("transactions error: {0}")]
+    Transactions(#[from] crate::TransactionsError),
+    #[error("ws event error: {0}")]
+    WsEvent(#[from] crate::WsEventError),
+}
+
+pub type Result<T> = std::result::Result<T, PromptOverrideError>;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct PromptUpdatedPayload {
+    pub kind: String,
+    pub overridden: bool,
+}
+
+impl WsEvent {
+    pub async fn prompt_updated(
+        ctx: &DalContext,
+        kind: String,
+        overridden: bool,
+    ) -> WsEventResult<Self> {
+        WsEvent::new(
+            ctx,
+            WsPayload::PromptUpdated(PromptUpdatedPayload { kind, overridden }),
+        )
+        .await
+    }
+}
+
+pub struct PromptOverride;
```

Comment:
Scoping things by a unit struct is a great idea in this instance because it appears that we do not want to return the entire row, but rather the prompt override itself, which is "just" a string. Is that right?

Reply:
Yep. It felt better having a unit struct to scope in this instance for `PromptOverride::get` rather than `prompt_override::get` or `prompt_override::get_prompt_override`.